### DISCUSSION TITLE
How to set SAN:othername

### DISCUSSION THREAD

**Discussion by burnbabyburn** [smallstep/certificates#430] [Q&A] [answered]
I am trying to create Samba AD-DC certs with smallstep which [requires](https://docs.microsoft.com/en-us/troubleshoot/windows-server/windows-security/requirements-domain-controller):
setting [SAN:OtherName](https://www.alvestrand.no/objectid/2.5.29.17.html)

looks like this in openssl for [samba AD-DC](https://wiki.samba.org/index.php/Samba_AD_Smart_Card_Login#User_Configuration_and_Directory_Organization)
```
[ new_oids ]
msADGUID=1.3.6.1.4.1.311.25.1

[dc_subjalt]
DNS=[DNS name of domain controller]
otherName=msADGUID;FORMAT:HEX,OCTETSTRING:[DC's GUID in HEX, eg, FEEDDEADBEEF...]
```
We can set eku with unknownkeyextension and define custom oids in template
But how to set SAN OtherName?
Can [smallstep ](https://pkg.go.dev/go.step.sm/crypto/x509util#SplitSANs) parse it?
`SplitSANs(sans []string) (dnsNames []string, ips []net.IP, emails []string, uris []*url.URL)`

IF #206 gets resolved in the future, one could issue certificates for Samba AD/KDC Server and client SmardCard login. (e.g. from a TPM based virtuell SC on W10) 
Which would be nice

**Comment by maraino** [comment, marked as answer]
Currently, the only way to set the `otherName` SAN would be to add the raw subject alternative extension as an `Extension` in the template but to do that you need to manually marshal the full extension value.

**Reply by maraino** [reply]
@burnbabyburn can you create an issue to support otherName as well as the rest of SANs using templates.

**Reply by maraino** [reply]
This question is now tracked in https://github.com/smallstep/certificates/issues/431
